From 60e4b9e8eeed9f8100fe92292ff3c013b8a54637 Mon Sep 17 00:00:00 2001
Message-Id: <60e4b9e8eeed9f8100fe92292ff3c013b8a54637.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Mon, 19 Jan 2015 10:44:07 +0530
Subject: [PATCH 90/95] MUST_REBASE [VPG]: drm/i915: Allow DPST register
 access when system in suspend

When system was in suspend and when we tried to read DPST status, we end
up reading wrong value & resulting in showing invalid data which is
causing GFT test it fail.

Since DPST registers is part Display unit, forcewake the Display unit to
read registers in proper state. Only Display 2D will be power up not the
panel, after the register read system goes back to suspend.

MUST_REBASE: DPST will not be upstreamed.

Issue: GMIN-3624
Change-Id: Ie5d62c15f9b5f3f78b8a43f0b58ac2216278e19d
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 2ffd225..f9ba549 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -1705,6 +1705,7 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 
 	mutex_lock(&dev_priv->dpst.ioctl_lock);
 
+	intel_display_power_get(dev_priv, PIPE_A);
 	blm_hist_ctl = I915_READ(dev_priv->dpst.reg.blm_hist_ctl);
 	blm_hist_guard = I915_READ(dev_priv->dpst.reg.blm_hist_guard);
 
@@ -1734,6 +1735,7 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 		seq_printf(m, "%10x", I915_READ(dev_priv->dpst.reg.blm_hist_bin));
 	}
 	seq_puts(m, "\n");
+	intel_display_power_put(dev_priv, PIPE_A);
 
 	mutex_unlock(&dev_priv->dpst.ioctl_lock);
 
-- 
1.7.9.5

